app/src/main/kotlin/com/softteco/template/ui/feature/signUp/SignUpViewModel.kt

```diff
@@ -40,14 +40,16 @@ class SignUpViewModel @Inject constructor(
 
     val state = combine(
         loading,
+        registrationState,
         userNameStateValue,
         emailStateValue,
         passwordStateValue,
         snackBarState,
         emailFieldState
-    ) { loading, userName, emailValue, passwordValue, snackBar, emailState ->
+    ) { loading, registrationState, userName, emailValue, passwordValue, snackBar, emailState ->
```

Comment:
This is a flag that we need to optimize our State model.
For example, add a class that will provide the state of a request or some function action:
```
sealed class SignupState {
    object Loading : SignupState()
    object Success : SignupState()
    class Error(val snackbar: SnackBarState) : SignupState()
}
```